scripts/mc_launch_rtp.py

```diff
@@ -180,29 +187,49 @@ def _get_obsids(filelist):
         filelist, args.workflow_config, mf_filename, jd_folder
     )
 
-    # update RTP launch records
-    obsids = _get_obsids(filelist)
-    t0 = Time.now()
-    for filename, obsid in zip(filelist, obsids):
-        try:
-            mc.update_rtp_launch_record(obsid, t0)
-        except RuntimeError:
-            sys.exit(
-                f"could not update RTP Launch Record for file {filename}; aborting"
-            )
-
     # launch workflow inside of tmux
+    # use screen instead of tmux until we can update tmux
+    # cmd = (
+    #     f"conda deactivate; conda activate {args.conda_env}; "
+    #     f"makeflow -T slurm {mf_filename}"
+    # )
+    # session_name = f"rtp_{jd}"
+    # tmux_cmd = ["tmux", "-d", "-s", session_name, cmd]
+    # try:
+    #     subprocess.check_call(tmux_cmd)
+    # except subprocess.CalledProcessError as e:
+    #     sys.exit(
+    #         f"Error spawning tmux session; command was {e.cmd}; "
+    #         f"returncode was {e.returncode:d}; output was {e.output}; "
+    #         f"stderr was {e.stderr}"
+    #     )
     cmd = (
         f"conda deactivate; conda activate {args.conda_env}; "
-        f"makeflow -T slurm {mf_filename}"
+        f"makeflow -T slurm {mf_filename}\n"
     )
-    session_name = f"rtp_{jd}"
-    tmux_cmd = ["tmux", "-d", "-s", session_name, cmd]
+    screen_name = f"rtp_{jd}"
+    screen_cmd1 = ["screen", "-d", "-m", "-S", screen_name]
+    screen_cmd2 = ["screen", "-S", screen_name, "-p", "0", "-X", "stuff", cmd]
     try:
-        subprocess.check_call(tmux_cmd)
+        subprocess.check_call(screen_cmd1)
+        subprocess.check_call(screen_cmd2)
     except subprocess.CalledProcessError as e:
         sys.exit(
-            f"Error spawning tmux session; command was {e.cmd}; "
-            f"returncode was {e.returncode:d}; output was {e.output}; "
-            f"stderr was {e.stderr}"
+            f"Error spawning screen session; "
+            f"command was {e.cmd}; return code was {e.returncode:d}; "
+            f"output was {e.output}; stderr was {e.stderr}"
         )
+
+    # update RTP launch records
+    if len(obsids) == 0:
+        obsids = _get_obsids(filelist)
+    t0 = Time.now()
+    with db.sessionmaker() as session:
+        for filename, obsid in zip(filelist, obsids):
+            try:
+                session.update_rtp_launch_record(obsid, t0)
+            except RuntimeError:
+                sys.exit(
+                    f"could not update RTP Launch Record for file {filename}; aborting"
+                )
+    session.commit()
```

Comment:
Two things concerning this commit:
- It is outside the session context
- I'm not sure it is necessary either.
`update_rtp_launch_record` has a `self.commit()` before returning and if this is only used to make sure the launch records are committed then I think it is redundant. Worth checking out.

Also the `__exit__` method calls `commit` by default as long as there is not an error.

Reply:
Oh there is another `session.commit()` call in the other file that was edited too. Same comment there. I must not have saved my comment before posting.

Reply:
Thanks for pointing this out. This is fixed in 5d45070.